lib/active_shipping/carriers/ups.rb

```diff
@@ -816,16 +816,18 @@ def parse_tracking_response(response, options = {})
         # Build status hash
         status_nodes = first_package.css('Activity > Status > StatusType')
 
-        # Prefer a delivery node
-        status_node = status_nodes.detect { |x| x.at('Code').text == 'D' }
-        status_node ||= status_nodes.first
+        if status_nodes.present?
```

Comment:
We have an existing 💩  pattern of doing this, `nil` is a known possible result of a `status`. 

[USPS does the exact same thing](https://github.com/Shopify/active_shipping/blob/master/lib/active_shipping/carriers/usps.rb#L636)
[FedEx](https://github.com/Shopify/active_shipping/blob/master/lib/active_shipping/carriers/fedex.rb#L632) doesn't validate that the status code isn't `nil`.